### PR TITLE
Show error and warning indicators in tabs

### LINKED ISSUE #21179: Show error and warning indicators in tabs
### Check for existing issues

- [X] Completed

### Describe the feature

In https://github.com/zed-industries/zed/pull/18182, we introduced diagnostics to the project panel using a `DecoratedIcon` with X or triangle decorations for errors and warnings. It would be great to also show diagnostics in the tab bar as well, using the same approach. I would also use the same setting we used in the `"project_panel"`.

```json
{
  "tabs": {
    // "all" / "errors" / "off"
    "show_diagnostics": "all"
  }
}
```

I am happy to implement this feature if you would want to add this to Zed.

### Environment

Zed: v0.163.2 (Zed Preview)
OS: macOS 15.1.1
Memory: 18 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help present your vision of the feature

tabs

### PR DESCRIPTION
**Pull request by nilskch**
Closes #21179

Release Notes:

- Add setting to display error and warning indicators in tabs.

demo_with_icons
demo_without_icons

### CODE REVIEW

**Comment by nilskch** [comment]
Even though I didn't get feedback on the related issue yet, I wanted to take a proactive approach and offer an implementation for consideration. This PR is intended as a starting point for discussion, not a push for immediate adoption. I think this would be a nice addition to https://github.com/zed-industries/zed/pull/18182. Feedback is very welcome!

File: crates/workspace/src/pane.rs [hunk 1]
```diff
@@ -1841,23 +1886,6 @@ impl Pane {
         }
     }
 
-    pub fn git_aware_icon_color(
```
Comment (mikayla-maki) [inline crates/workspace/src/pane.rs:1844]: Does this PR entirely remove the implementation of `tabs.git_status`?

**Comment by mikayla-maki** [comment]
Could you include a before image in your example?

**Comment by nilskch** [comment]
| Before         | After          |
|----------------|----------------|
| before | after |


It is important to note that if you set `tabs.show_diagnostics = off`, we still do not change the icon color with the git_status after this PR:


off


I did this to make it consistent with the project panel after the changes in https://github.com/zed-industries/zed/issues/21179:

|`project_panel.show_diagnostics = all`         | `project_panel.show_diagnostics = off`          |
|----------------|----------------|
| project_panel_all |project_panel_off |

File: crates/workspace/src/pane.rs (hunk 1 above)
Comment (nilskch) [reply crates/workspace/src/pane.rs:1842]: No it only removes the coloring of the icons depending on the git status. The text color still changes with the git status if `tabs.git_status = true`. https://github.com/zed-industries/zed/issues/21179 did the same in the project_panel.

**Review by danilo-leal — APPROVED** [review]
Sweet, this looks great to me, thanks so much!

**Pull request merged**